builtin/submodule--helper.c

```diff
@@ -1214,9 +1214,7 @@ static void submodule_summary_callback(struct diff_queue_struct *q,
 
 		if (!S_ISGITLINK(p->one->mode) && !S_ISGITLINK(p->two->mode))
 			continue;
-		temp = (struct module_cb*)malloc(sizeof(struct module_cb));
-		if (!temp) 
-			die(_("out of memory"));
+		temp = (struct module_cb*)xmalloc(sizeof(struct module_cb));
```

Comment:
You probably want to squash these two commits together, using the second commit message (because it wraps the text correctly).